articles/regularization/index.md

```diff
@@ -0,0 +1,84 @@
+
+$$ \sum\limits_{i=1}^n( y_i – \beta_0 - \sum\limits_{j=1}^p {\beta_j x_ij )^2} + \lambda \sum\limits_{j=1}^p \beta_j^2 = RSS + \lambda \sum\limits_{j=1}^p \beta_j^2 $$
+
+Above is the ridge regression equation. AS we can see, the difference between the above equation and the general equation for the loss function is that $ \lambda \sum\limits_{j=1}^p \beta_j^2 $ is added on both sides. A shrinkage quantity is added to RSS, thus modifying it. To estimate the coefficients, the function above has to be minimized. A penalty term is introduced. To penalize the flexibility of our model, we use a tuning parameter that decides the extent of the penalty. The tuning parameter is $\lambda$. Higher values of the coefficients represent a model with greater flexibility. To minimize the function, these coefficients should be small. L2 regularization ensures the coefficients do not rise too high.
+
+As can be noted from the equation, if the value of $\lambda$ tends to zero, the equation becomes the loss function of the linear regression model. The penalty term will have no effect. This means that if the value of $\lambda$ is at a minimum, the model will be similar to a linear regression model. Conversely, if the $\lambda$ value tends to infinity, the effect of the shrinkage penalty increases. As a result, the coefficient estimates of the ridge regression will approach zero. This underlines the importance of choosing a good value of $\lambda$.
+
+Linear or polynomial regression will likely prove unsuccessful if there is high [collinearity](https://machinelearningmind.com/2019/10/19/multicollinearity-how-to-fix-it/) between the independent variables. Ridge regression is a potential solution to handle multicollinearity. We shall explore this in a different article.
+
+However, ridge regression has one clear disadvantage; model interpretability. When shrinking the coefficients of the predictors of the least importance, it will get them to be very close to zero. Here’s the problem. They won’t actually be zero. As a result, one may end up including all the coefficients in the final model. Whereas lasso regression manages to force some coefficient estimates to zero when $\lambda$ is large enough. Resultantly, we can say lasso performs feature selection. Furthermore, the resulting models can be considered sparser compared to the ones resulting from the L2 technique. This is because some of the estimate coefficients can be truly zero. The number of predictors is reduced.
+
+### Wrapping Up
+
+Regularization is a formidable technique to prevent overfitting. There are several regularization techniques. We have looked at two of the most popular ones. Through this article, we have understood how the two compare to each other and how they are derived from the linear regression loss function equation. Until next time, good luck!
```

Comment:
```suggestion
Regularization is a formidable technique to prevent overfitting. There are several regularization techniques. We have looked at two of the most popular ones. Through this article, we have understood how the two techniques help in preventing overfitting and how they are derived from the linear regression loss function equation. Until next time, good luck!
```